### PR TITLE
 [ci] Add unit test for banned keywords, modernize some code

### PR DESCRIPTION
**Pull request by nyalldawson**
Flags use of keywords representing deprecated or other "to be avoided" methods, e.g. use of DBL_MAX instead of the type-safe std::numeric_limits<double>::max() method.

### CODE REVIEW

File: tests/code_layout/test_banned_keywords.sh [hunk 1]
```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
```
Comment (m-kuhn) [inline tests/code_layout/test_banned_keywords.sh]: IIRC it was recommended to use `/usr/bin/sh` to avoid "bashism"

**Comment by m-kuhn** [comment]
Sweet

File: tests/code_layout/test_banned_keywords.sh (hunk 1 above)
Comment (nyalldawson) [reply tests/code_layout/test_banned_keywords.sh]: Actually every other .sh file uses `/usr/bin/env bash`, including all those from the code_layout tests

Comment (jef-n) [reply tests/code_layout/test_banned_keywords.sh]: `sh` lives in `/bin` anyway.   And a bashism is (sloppy) use of a bash extension in script meant to be executed with `/bin/sh` - which will fail if `/bin/sh` is not a bash - which is used to be on Linux for long.   So everything fine - bash should be intentional as `test_banned_keywords.sh` uses bash extensions (`declare`, arrays…)

**Pull request merged**